tests/rabbitmq.rs

```diff
@@ -0,0 +1,124 @@
+use tokio_amqp::LapinTokioExt;
+
+#[tokio::test]
+async fn rabbitmq_produce_and_consume_messages() -> Result<(), Error> {
+    let _ = pretty_env_logger::try_init()?;
+    let docker = clients::Cli::default();
+    let rabbit_node = docker.run(rabbitmq::RabbitMq::default());
+
+    let amqp_url = format!("amqp://localhost:{}", rabbit_node.get_host_port(5672));
+
+    let connection = Connection::connect(
+        amqp_url.as_str(),
+        ConnectionProperties::default().with_tokio(),
+    )
+    .await?;
+
+    let channel = connection.create_channel().await?;
+
+    channel
+        .exchange_declare(
+            "test_exchange",
+            ExchangeKind::Topic,
+            ExchangeDeclareOptions::default(),
+            FieldTable::default(),
+        )
+        .await?;
+
+    let queue = channel
+        .queue_declare(
+            "test_queue",
+            QueueDeclareOptions::default(),
+            FieldTable::default(),
+        )
+        .await?;
+
+    channel
+        .queue_bind(
+            queue.name().as_str(),
+            "test_exchange",
+            "#",
+            QueueBindOptions::default(),
+            FieldTable::default(),
+        )
+        .await?;
+
+    let mut consumer = channel
+        .basic_consume(
+            queue.name().as_str(),
+            "test_consumer_tag",
+            BasicConsumeOptions::default(),
+            FieldTable::default(),
+        )
+        .await?;
+
+    channel
+        .basic_publish(
+            "test_exchange",
+            "routing-key",
+            BasicPublishOptions::default(),
+            b"Test Payload".to_vec(),
+            BasicProperties::default(),
+        )
+        .await?;
+
+    let consumed = tokio::time::timeout(Duration::from_secs(10), consumer.next())
+        .await?
+        .ok_or(Error {
+            message: "Nothing consumed.".to_string(),
+        })?;
+    let (_, delivery) = consumed.expect("Failed to consume delivery!");
+    assert_eq!(String::from_utf8(delivery.data.clone())?, "Test Payload");
+    assert_eq!(delivery.exchange.as_str(), "test_exchange");
+    assert_eq!(delivery.routing_key.as_str(), "routing-key");
+
+    Ok(())
+}
+
+#[derive(Debug)]
+struct Error {
+    message: String,
+}
+
+impl From<lapin::Error> for Error {
+    fn from(error: lapin::Error) -> Self {
+        Error {
+            message: error.to_string(),
+        }
+    }
+}
+
+impl From<log::SetLoggerError> for Error {
+    fn from(error: log::SetLoggerError) -> Self {
+        Error {
+            message: error.to_string(),
+        }
+    }
+}
+
+impl From<std::string::FromUtf8Error> for Error {
+    fn from(error: std::string::FromUtf8Error) -> Self {
+        Error {
+            message: error.to_string(),
+        }
+    }
+}
+
+impl From<tokio::time::error::Elapsed> for Error {
+    fn from(error: tokio::time::error::Elapsed) -> Self {
+        Error {
+            message: error.to_string(),
+        }
+    }
+}
```

Comment:
Instead of returning `Result` from the test, please use `.unwrap` directly. Not only does this give us line numbers, it is also a lot shorter :)

Reply:
> > @thomaseizinger Regarding ownership of the image (see #298), is there anything I should do communicate the ownership?
> 
> No that is fine. I can track back based on the Git history! Although if you'd want to create a document that tracks the ownership, that would be great! :)

What do you think about using a [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) file for that?

Reply:
Good point, I changed that 👍 
Thanks for the explanation :)